.goreleaser.yaml

```diff
@@ -39,12 +41,23 @@ archives:
 checksum:
   name_template: 'checksums.txt'
 snapshot:
-  name_template: "{{ incpatch .Version }}-next"
+  version_template: "{{ incpatch .Version }}-next"
 changelog:
   sort: asc
   filters:
     exclude:
       - '^docs:'
       - '^test:'
       - '^ci:'
-
+signs:
+  - cmd: cosign
+    signature: "${artifact}.sig"
+    certificate: "${artifact}.pem"
+    args:
+      - "sign-blob"
+      - "--oidc-issuer=https://token.actions.githubusercontent.com"
```

Comment:
Does this GitHub OIDC issuer URL mean that a OIDC token is requested on an action run, and then that token is what is used by cosign to sign the artifact produced by this particular run of the releasing action?

Reply:
Yes, that's correct. There are probably two ways of doing this; I chose specifying the issuer directly via `--oidc-issuer`. From `cosign sign --help`:

```
    --oidc-issuer='https://oauth2.sigstore.dev/auth':
        OIDC provider to be used to issue ID token

    --oidc-provider='':
        Specify the provider to get the OIDC token from (Optional). If unset, all options will be tried. Options
        include: [spiffe, google, github-actions, filesystem, buildkite-agent]
```

Then, on the verification side, `--certificate-identity-regexp` is mandatory to verify the certificate.

```bash
cosign verify-blob checksums.txt \
   --certificate checksums.txt.pem \
   --signature checksums.txt.sig \
   --certificate-oidc-issuer "https://token.actions.githubusercontent.com"

# Error: --certificate-identity or --certificate-identity-regexp is required for verification in # keyless mode
# main.go:74: error during command execution: --certificate-identity or --certificate-identity-regexp is required for verification in keyless mode
```

```bash
cosign verify-blob checksums.txt \
   --certificate checksums.txt.pem \
   --signature checksums.txt.sig \
   --certificate-identity-regexp 'https://github\.com/dhth/yamlfmt/\.github/workflows/.+' \
   --certificate-oidc-issuer "https://token.actions.githubusercontent.com"

# Verified OK
```

Reply:
Thanks for the explanation!